STYLE_GUIDE.md

```diff
@@ -0,0 +1,88 @@
+# Style Guide
+
+This document outlines the style guide for the `epinowcast` package. This guide is a work in progress and will be updated as the package evolves. We welcome contributions to this guide and encourage you to raise issues or submit PRs if you have any suggestions.
+
+In addition to this guide we also follow the [tidyverse style guide](https://style.tidyverse.org/). This guide is a subset of the `tidyverse` style guide and outlines the additional style requirements for the `epinowcast` package.
+
+## Naming conventions
+
+- We use a `enw_` prefix to delineate functions that are part of the `epinowcast` package. This is to avoid conflicts with other packages and to make it clear to users which functions are part of the package.
+- The use of this prefix is not required for internal functions (i.e. functions that are not exported) or that are unlikely to have naming conflicts with other packages.
+
```

Comment:
Feels like there are some other naming conventions here - e.g. `_add_` meaning that the submitted data.frame is modified? Perhaps ones that we want to, but do not yet, enforce.

Reply:
any other vocabulary we want to define here? e.g. `rw` meaning random walk, `re` meaning...random effect?

Reply:
The add family could be flagged but it doesn't seem essential. I think pulling out the rw and re might be a bit detailed for this style guide?

Reply:
perhaps, was mostly proposing those as examples. how about the use of "date" as per the other thread?

my point here is that seems like there's a bit of nascent vocabulary and that it would be worthwhile to codify that *a bit* (obviously, can evolve as needed) so that we proceed intentionally with function / argument name choices.

Reply:
I think we need to discuss this in an issue/discussion. Could you take point on that? Resolving for this PR.